proposal/color-4-new-spaces-js.d.ts.md

```diff
@@ -0,0 +1,540 @@
+  * [New Constructors](#new-constructors)
+    * [RGB Channel Constructor](#rgb-channel-constructor)
+    * [HSL Channel Constructor](#hsl-channel-constructor)
+    * [HWB Channel Constructor](#hwb-channel-constructor)
+    * [XYZ Channel Constructor](#xyz-channel-constructor)
+    * [LCH Channel Constructor](#lch-channel-constructor)
+    * [LAB Channel Constructor](#lab-channel-constructor)
+  * [Deprecations](#deprecations)
+* [Embedded Protocol](#embedded-protocol)
+  * [SassColor](#sasscolor)
+  * [Deprecations](#deprecations-1)
+
+## API
+
+```ts
+import {Value} from '../spec/js-api/value';
+```
+
+## Types
+
+### Color Space Definitions
+
+```ts
+type ColorSpaceRGB =
+  | 'rgb'
+  | 'srgb'
+  | 'srgb-linear'
+  | 'display-p3'
+  | 'a98-rgb'
+  | 'prophoto-rgb';
+
+type ChannelNameRGB = 'red' | 'green' | 'blue';
+
+type ColorSpaceHWB = 'hwb';
+
+type ChannelNameHWB = 'hue' | 'whiteness' | 'blackness';
+
+type ColorSpaceHSL = 'hsl';
+
+type ChannelNameHSL = 'hue' | 'saturation' | 'lightness';
+
+type ColorSpaceXYZ = 'xyz' | 'xyz-d50' | 'xyz-d65';
+
+type ChannelNameXYZ = 'x' | 'y' | 'z';
+
+type ColorSpaceLCH = 'lch' | 'oklch';
+
+type ChannelNameLCH = 'lightness' | 'chroma' | 'hue';
+
+type ColorSpaceLAB = 'lab' | 'oklab';
+
+type ChannelNameLAB = 'lightness' | 'a' | 'b';
+
+type ChannelName =
+  | ChannelNameRGB
+  | ChannelNameHWB
+  | ChannelNameHSL
+  | ChannelNameXYZ
+  | ChannelNameLCH
+  | ChannelNameLAB;
+
+type KnownColorSpace =
+  | ColorSpaceRGB
+  | ColorSpaceHWB
+  | ColorSpaceHSL
+  | ColorSpaceXYZ
+  | ColorSpaceLCH
+  | ColorSpaceLAB;
+
+type PolarColorSpace =
+  | ColorSpaceHWB
+  | ColorSpaceHSL
+  | ColorSpaceLCH
+  | ColorSpaceLAB;
+
+type RectangularColorSpace = Omit<KnownColorSpace, PolarColorSpace>;
+
+type HueInterpolationMethod =
+  | 'shorter'
+  | 'longer'
+  | 'decreasing'
+  | 'increasing';
+
+type ChannelValue = string | number | null;
+```
+
+### New Color Functions
+
+```ts
+export class SassColor extends Value {
+```
+
+#### `space`
+
+Returns the name of [`internal`]'s space.
+
+```ts
+get space(): KnownColorSpace;
+```
+
+#### `channels`
+
+Returns an array of channel values for [`internal`], with missing channels
+converted to `0`.
+
+```ts
+get channels(): string | number;
+```
+
+#### `channelsOrNull`
+
+Returns an array of channel values for [`internal`], with missing channels
+converted to `null`.
+
+```ts
+get channelsOrNull(): ChannelValue;
+```
+
+#### `isLegacy`
+
+Returns whether [`internal`] is in a legacy color space (`rgb`, `hsl`, or
+`hwb`).
+
+```ts
+get isLegacy(): boolean;
+```
+
+#### `isInGamut`
+
+Returns whether [`internal`] is in-gamut for its color space (as opposed to
+having one or more of its channels out of bounds, like `rgb(300 0 0)`).
+
+```ts
+get isInGamut(): boolean;
+```
+
+#### `channel`
+
+Returns the value of the given `channel` in [`internal`], after converting it to
+`space` if necessary. It should be used instead of the old channel-specific
```

Comment:
As above, you need to be explicit about what it means to convert a color to a space (and again you can just link to the color spec). In this case, "if necessary" is unnecessary, since the "Converting a Color" procedure already short-circuits if the color is in the target space.